### PR TITLE
Uia

### PR DESCRIPTION
**Pull request by cetygamer**

### CODE REVIEW

File: dev-requirements.txt [hunk 1]
```diff
@@ -2,4 +2,6 @@ pypiwin32
 Pillow==3.0.0
 coverage
 nose
-codecov
\ No newline at end of file
+codecov
+rst2pdf
+Sphinx
```
Comment (airelil) [inline dev-requirements.txt]: Does a user really need rst2pdf & Sphinx ? Aren't they just for developers?

Comment (vasily-v-ryabov) [inline dev-requirements.txt]: The file name is **dev-**requirements.txt. :) This is my experiments with pdf format of the docs. Sphinx html docs is a build artifact now. Pdf docs still fail.

File: pywinauto/controls/BaseWrapper.py [hunk 2]
```diff
@@ -0,0 +1,816 @@
...
+
+    # Properties that describe type of the element
+    can_be_label = False
+    has_title = True
+
+    #------------------------------------------------------------
+    def __new__(cls, elementInfo):
+        # only use the meta class to find the wrapper for BaseWrapper
+        # so allow users to force the wrapper if they want
+        if cls != BaseWrapper:
```
Comment (airelil) [inline pywinauto/controls/BaseWrapper.py]: ```if cls is not BaseWrapper:``` is a bit cleaner (some say it even faster) 
http://stackoverflow.com/questions/132988/is-there-a-difference-between-and-is-in-python
http://jaredgrubb.blogspot.co.nz/2009/04/python-is-none-vs-none.html

File: dev-requirements.txt (hunk 1 above)
Comment (airelil) [inline dev-requirements.txt]: Ok. I didn't notice it. My mistake.

File: pywinauto/backend.py [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    Set active back-end by name
+
+    Possible values of **active_name** are "native", "uia" or
+    other name registered by the **register** function.
+    """
+    if new_active_name not in registered_backends.keys():
+        raise ValueError('Back-end "{backend}" is not registered!'.format(backend=new_active_name))
+
+    #global active_name, ActiveElementInfo, ActiveWrapper
+    global active_name, ActiveElementInfo
```
Comment (airelil) [inline pywinauto/backend.py]: QuantifiedCode already complaints about it. Creating a class would be better. Something like this:
```
class Cfg(object):
    active_name = None
    ActiveElementInfo = None

    @classmethod
    def activate(new_active_name):
        ....
cfg = Cfg()
```

File: pywinauto/__init__.py [hunk 4]
```diff
@@ -29,7 +29,12 @@
 
 from . import findwindows
 WindowAmbiguousError = findwindows.WindowAmbiguousError
-WindowNotFoundError = findwindows.WindowNotFoundError
+ElementNotFoundError = findwindows.ElementNotFoundError
+
+from .sysinfo import UIA_support
+if UIA_support:
+    ElementNotFoundError = findwindows.ElementNotFoundError
```
Comment (airelil) [inline pywinauto/__init__.py]: This statement is a duplicate to the line 32

File: pywinauto/UIAElementInfo.py [hunk 5]
```diff
@@ -20,38 +28,89 @@
...
-        """Create instane of UIAElementInfo from handle.
-        If handle is None create instanse for UI root element"""        
-        if handle is not None:
-            self._element = _iuia.ElementFromHandle(handle)
+    def __init__(self, handle_or_elem = None):
+        """
+        Create instane of UIAElementInfo from a handle (int or long)
+        or from an IUIAutomationElement.
+        If handle_or_elem is None create instanse for UI root element
+        """
```
Comment (airelil) [inline pywinauto/UIAElementInfo.py]: Typo errors. Should be "instance", not instane/instanse

**Pull request merged**